File: backend/model/activity.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Activity struct {
	ID           int            `gorm:"primaryKey;autoIncrement" json:"id"`
	ActivityUUID string         `gorm:"column:activity_uuid;unique;not null" json:"activity_uuid"`
	Title        string         `gorm:"column:title;not null" json:"title"`
	Type         string         `gorm:"column:type;not null" json:"type"` // call, meeting, email, note
	Description  string         `gorm:"column:description" json:"description"`
	RelatedType  string         `gorm:"column:related_type;not null" json:"related_type"` // lead, deal, contact
	RelatedID    int            `gorm:"column:related_id;not null" json:"related_id"`
	ScheduledAt  time.Time      `gorm:"column:scheduled_at" json:"scheduled_at"`
	Status       string         `gorm:"column:status;default:'pending'" json:"status"` // pending, completed
	OwnerID      *int           `gorm:"column:owner_id" json:"owner_id"`
	CreatedAt    time.Time      `gorm:"column:created_at" json:"created_at"`
	UpdatedAt    time.Time      `gorm:"column:updated_at" json:"updated_at"`
	DeletedAt    gorm.DeletedAt `gorm:"column:deleted_at" json:"deleted_at"`
}

func (Activity) TableName() string {
	return "activities"
}

File: frontend/src/utils/dto/activity.ts
export enum ActivityType {
    CALL = 'call',
    MEETING = 'meeting',
    EMAIL = 'email',
    NOTE = 'note'
}

export enum ActivityRelatedType {
    LEAD = 'lead',
    DEAL = 'deal',
    CONTACT = 'contact'
}

export enum ActivityStatus {
    PENDING = 'pending',
    COMPLETED = 'completed'
}

export interface ActivityResponse {
    id: number;
    activity_uuid: string;
    title: string;
    type: ActivityType;
    description: string;
    related_type: ActivityRelatedType;
    related_id: number;
    scheduled_at: string;
    status: ActivityStatus;
    owner_id?: number;
    created_at: string;
    updated_at: string;
}

export interface ActivityListRequest {
    page: number;
    limit: number;
    start_date?: string;
    end_date?: string;
    status?: string | 'All';
    type?: string | 'All';
    related_type?: string | 'All';
    owner_id?: number;
}

export interface ActivityCreateRequest {
    title: string;
    type: ActivityType;
    description?: string;
    related_type: ActivityRelatedType;
    related_id: number;
    scheduled_at?: string;
    status?: ActivityStatus;
    owner_id?: number;
}

export interface ActivityUpdateRequest {
    id: number;
    title?: string;
    type?: ActivityType;
    description?: string;
    scheduled_at?: string;
    status?: ActivityStatus;
}
